src/Storages/MergeTree/DataPartStorageOnDiskBase.cpp

```diff
@@ -229,7 +229,7 @@ bool DataPartStorageOnDiskBase::isBroken() const
 
 bool DataPartStorageOnDiskBase::isReadonly() const
 {
-    return volume->getDisk()->isReadOnly();
```

Comment:
May be add a separate method `DataPartStorageOnDiskBase::isWriteOnce` and check it in `loadColumns` instead of here? Seems safer

Reply:
I don't see that isReadOnly used anywhere, and besides, the write-most disk is almost as read-only disk right now in terms of MergeTree, since any writing there is done with rename

Bu I'm to rename it to isReadOnlyOrWriteOnce, though looks a little too verbose to me

Reply:
Ok, I do not have a strong opinion here as well, it just seems more transparent if we explicitly check for certain type (readOnly, writeOnce) to make sure we do not accidentally break something in the future when we try to use this method somewhere else.